### PR TITLE
Revamping "Introduction to Qiskit Serverless"

### LINKED ISSUE #1639: [ NEW PAGE ] Intro to Qiskit Serverless
This page should include:
- What is Qiskit Serverless, QCSC, etc.
- when to user serverless vs batch/sessions
- Install instructions

You can take content from the following existing pages:
- [Qiskit Serverless Homepage](https://qiskit.github.io/qiskit-serverless/)
- [Installation](https://qiskit.github.io/qiskit-serverless/installation/index.html)

### PR DESCRIPTION
**Pull request by pandasa123**
## Description

This is the first serverless docs update to update the Introduction to Qiskit Serverless.

Closes #1639.

### CODE REVIEW

File: docs/guides/qiskit-serverless.mdx [hunk 1]
```diff
@@ -1,232 +1,32 @@
...
+description: Use Qiskit Serverless to run quantum-classical workloads across cloud and quantum-centric supercomputing environments.
 ---
 
-# Run workloads remotely with Qiskit Serverless
+# What is Qiskit Serverless?
 
-Premium users can build, deploy, and run their workloads remotely on classical compute made available through the IBM Quantum&trade; Platform.
-
-Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) (note: these are accessible in the Premium Plan once you have logged into your IBM Quantum account) and explore more of the features of Qiskit Serverless in the [documentation](https://qiskit.github.io/qiskit-serverless/).
+Qiskit Serverless provides a simple interface to run workloads across quantum-classical resources. This includes deploying programs to the IBM Quantum Platform, running workloads remotely, and easy resource management across cloud and quantum-centric supercomputing environments.
```
Comment (abbycross) [inline docs/guides/qiskit-serverless.mdx:8]:
```suggestion
Qiskit Serverless provides a simple interface to run workloads across quantum-classical resources. Capabilities include deploying programs to IBM Quantum&trade; Platform, running workloads remotely, and easy resource management across cloud and quantum-centric supercomputing environments.
```

File: docs/guides/qiskit-serverless.mdx [hunk 2]
```diff
@@ -1,232 +1,32 @@
...
 
+Premium users can build, deploy, and run their workloads remotely on classical compute made available through the IBM Quantum Platform. Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) (note: these are accessible in the Premium Plan once you have logged into your IBM Quantum account).
```
Comment (abbycross) [inline docs/guides/qiskit-serverless.mdx:10]:
```suggestion
Premium Plan users can build, deploy, and run their workloads remotely on classical compute made available through IBM Quantum Platform. Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) (note: these are accessible in the Premium Plan once you have logged into your IBM Quantum account).
```

File: docs/guides/qiskit-serverless.mdx [hunk 3]
```diff
@@ -1,232 +1,32 @@
...
+## When should I use Qiskit Serverless?
 
-Finally, the pattern is ready to run remotely.
+Qiskit Serverless is useful in the following situations: 
 
-```python
-# Run pattern remotely
-pattern = serverless.get("My-Qiskit-Pattern")
-pattern.run()
+- Parallelising classical tasks, such as pre-processing and post-processing
```
Comment (abbycross) [inline docs/guides/qiskit-serverless.mdx:20]:
```suggestion
- Parallelizing classical tasks, such as pre-processing and post-processing
```

File: docs/guides/qiskit-serverless.mdx [hunk 4]
```diff
@@ -1,232 +1,32 @@
...
-from qiskit.circuit.random import random_circuit
-from qiskit_serverless_tools.selectors import IBMLeastNoisyQPUSelector
-
-service = QiskitRuntimeService()
-
-# Step 1: Map quantum circuits and operators
-abstract_circuit = random_circuit(
-    num_qubits=5, depth=4, measure=True, seed=1234
-)
+## Installing Qiskit Serverless
```
Comment (abbycross) [inline docs/guides/qiskit-serverless.mdx:24]:
```suggestion
## Install Qiskit Serverless
```

File: docs/guides/qiskit-serverless.mdx [hunk 5]
```diff
@@ -1,232 +1,32 @@
...
-# save results of program execution
-# note: saved items must be serializable
-save_result(result)
-```
-
-  </TabItem>
-</Tabs>
 
-After creating this pattern, you can deploy and run it remotely with Qiskit Serverless as described above.
+This package requires `python` 3.10+. For users and organizations who want to run Serverless on custom infrastructure, follow the [deploy Qiskit Serverless on custom cloud resources guide](https://qiskit.github.io/qiskit-serverless/).
```
Comment (abbycross) [inline docs/guides/qiskit-serverless.mdx:32]: I don't see anything with that exact title - is it this? : https://qiskit.github.io/qiskit-serverless/deployment/cloud.html

File: docs/guides/qiskit-serverless.mdx [hunk 6]
```diff
@@ -1,232 +1,32 @@
 ---
-title: Run workloads remotely with Qiskit Serverless
-description: How to run quantum computing workloads remotely using Qiskit Serverless on IBM Quantum.
+title: Qiskit Serverless overview
```
Comment (javabster) [inline docs/guides/qiskit-serverless.mdx:2]: should this be "Introduction to Qiskit Serverless" to match the toc title?

File: docs/guides/qiskit-serverless.mdx [hunk 7]
```diff
@@ -1,232 +1,32 @@
...
-
-Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) (note: these are accessible in the Premium Plan once you have logged into your IBM Quantum account) and explore more of the features of Qiskit Serverless in the [documentation](https://qiskit.github.io/qiskit-serverless/).
+Qiskit Serverless provides a simple interface to run workloads across quantum-classical resources. This includes deploying programs to the IBM Quantum Platform, running workloads remotely, and easy resource management across cloud and quantum-centric supercomputing environments.
 
+Premium users can build, deploy, and run their workloads remotely on classical compute made available through the IBM Quantum Platform. Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) (note: these are accessible in the Premium Plan once you have logged into your IBM Quantum account).
 
 <Admonition type="note">
-    This is an experimental feature, subject to change.
+    This is a preview feature, subject to change.
```
Comment (javabster) [inline docs/guides/qiskit-serverless.mdx:13]: ugh I'm really not a fan of this "preivew" term, it means nothing to users, is there any way we can explain this term a bit more?

File: docs/guides/qiskit-serverless.mdx (hunk 5 above)
Comment (pandasa123) [reply docs/guides/qiskit-serverless.mdx:32]: Yep!

File: docs/guides/qiskit-serverless.mdx (hunk 6 above)
Comment (pandasa123) [reply docs/guides/qiskit-serverless.mdx:2]: Yep

File: docs/guides/qiskit-serverless.mdx (hunk 7 above)
Comment (pandasa123) [reply docs/guides/qiskit-serverless.mdx:13]: I think we can change how we phrase it, as long as we stay to the spirit of "preview"

Comment (abbycross) [reply docs/guides/qiskit-serverless.mdx]: How about something like "This feature is under continual development, subject to change" (or "frequent development")

**Comment by frankharkins** [comment]
@pandasa123 we've recently edited the history on `main`, can you please reset your branch using the following commands.

```sh
# Make sure you're on your branch
git switch serverless-refresh

# Overwrite your branch with our history
git reset --hard origin/main

# Re-apply your commits on top of the new history
git cherry-pick -X theirs 5d940bf~..cd8240b

# Check everything looks correct
git diff origin/main

# If all good, force-push to your branch
git push --force
```

Let me know if you have any questions. Sorry for the inconvenience.

**Comment by pandasa123** [comment]
Uh...going to close this PR and reopen

**Pull request closed without merging**